### DISCUSSION TITLE
IFreeSql.Ado.QueryAsync 的参数不支持  HashSet

### DISCUSSION THREAD

**Discussion by xunki** [dotnetcore/FreeSql#837] [Q&A]
会出现以下错误 
```
No mapping exists  from object type System.Collections.Generic.HashSet`1
```

目前将参数 .ToList() 或 .ToArray() 后可正常使用，是设计如此，还是暂不支持呢？

**Comment by luoyunchong** [comment]
object的那个重载支持  Dictionary 类型 、new{ id=1} 匿名类、new UserInfo{id=1} 具体的类实例。

其他的要看具体的代码解析这个object的代码。
https://github.com/dotnetcore/FreeSql/blob/6e5ef1e090/FreeSql/Internal/UtilsExpressionTree.cs#L1236


http://freesql.net/guide/ado.html

**Reply by xunki** [reply]
抱歉，应该是我描述不清楚。
我的参数是 `new { input.Ids }`  这里 `Ids` 如果是 `HashSet` 就会报错，转成 `Array` 或者 `List` 就没问题了。
之前用 Dapper 用习惯了，它是允许传入 `HashSet` 类型的，在 SQL 中当参数使用，像是 `WHERE Id IN @Ids`。

**Reply by luoyunchong** [reply]
上面链接写了。

> 当前仅支持Array和IList类型绑定
